hr_holidays_natural_period/models/resource_calendar.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime, time
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import models
+
+from odoo.addons.resource.models.resource import Intervals
+
+_logger = logging.getLogger(__name__)
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    def _attendance_intervals_batch(
+        self, start_dt, end_dt, resources=None, domain=None, tz=None
+    ):
+        intervals = super()._attendance_intervals_batch(
+            start_dt=start_dt, end_dt=end_dt, resources=resources, domain=domain, tz=tz
+        )
+        if self.env.context.get("natural_period"):
+            for resource in resources:
+                if resource:
+                    interval_resource = intervals[resource.id]
+                    if len(interval_resource._items) > 0:
+                        dt_item = start_dt.date()
+                        new_intervals = []
+                        while dt_item != end_dt.date():
+                            new_intervals.append(
+                                (
+                                    datetime.combine(dt_item, time.min).replace(
```

Comment:
I'm not sure about using this, as it can be conflicting due to timezone, showing holidays from previous/next days to the rest if in different zones. Real case: A leave from someone in Portugal for the day 2 of February, is shown in the calendar as being from 1 to 2 of February for Spanish HR managers.

Possibilities:

1. Put 12:00 as compromise solution (for the most extreme tz difference).
2. Take the first real calendar attendance at copy the starting hour.

The same can be apply to the ending hour.